### PR TITLE
experimental dev server

### PR DESCRIPTION
**Pull request by Deegiimurun**
### What's being changed:

Working on #2633. Need second opinion before finishing up.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: tools/dev/dev.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	path, _ := os.Getwd()
+
+	defaultEnvArgs := []string{
+		"GO111MODULE=on",
+		"LOG_LEVEL=debug",
+		"LOG_FORMAT=text",
+		"PERSISTENCE_DATA_PATH=./data",
+	}
+
+	if opts.Node == 1 {
```
Comment (dirkkul) [inline tools/dev/dev.go:112]: I think this is not an approach that scales- I'D rather do soemthing like

```
weavPort := 8080
gossipPort := 7100
clusterDataBindPort := 7101
for i in range node{
   *start node**
   weavPort += 1
   clusterDataBindPort +=2
   gossipPort += 2
}
```

Comment (Deegiimurun) [reply tools/dev/dev.go:112]: sgtm

**Comment by Deegiimurun** [comment]
hi @dirkkul, couldyou review this one?

File: tools/dev/dev.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+	"github.com/jessevdk/go-flags"
+)
+
+func main() {
+	var opts struct {
+		Node                   int    `long:"node" default:"1"`
+		Delete                 bool   `long:"delete"`
+		Restart                bool   `long:"restart"`
+		Keycloak               bool   `long:"keycloak"`
```
Comment (dirkkul) [inline tools/dev/dev.go:28]: what is this for? I don't see it in the original script?

File: tools/dev/dev.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/jessevdk/go-flags"
+)
+
+func main() {
+	var opts struct {
+		Node                   int    `long:"node" default:"1"`
```
Comment (dirkkul) [inline tools/dev/dev.go:25]: Ideally we could do "start 4 nodes" and it then spins up those 4 nodes. The config is exactly the same (besides the ports)

Comment (etiennedi) [reply tools/dev/dev.go:25]: The idea was to keep the script for exactly one node so you run it from 4 terminal tabs for example. This way, you can easily kill a node, restart a node. Switch the version of a node, etc. This helps debugging cluster issues.

Comment (dirkkul) [reply tools/dev/dev.go:25]: oki, then ignore :)

Comment (Deegiimurun) [reply tools/dev/dev.go:25]: @etiennedi could you give some feedback?

**Pull request closed without merging**